common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -264,6 +266,16 @@ public boolean onTouch(final View view, final MotionEvent event) {
         mWebView.setWebViewClient(webViewClient);
     }
 
+    /**
+     * NOTE: In the case that AADWebViewClient is using EoClientCertAuthChallengeHandler to handle CBA,
+     * the YubiKitManager needs to stop discovering Usb devices upon fragment destroy.
+     */
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        mAADWebViewClient.stopYubiKitManagerUsbDiscovery();
```

Comment:
Where/when do we start the discovery?

Reply:
This seems during IChallengeHandler constructor. I left comments there and we can continue the discussion there.

Reply:
In the constructor for EoClientCertAuthChallengeHandler: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1729/files#diff-75fd24e46d0e710da05bf7629216b41c67633fd3f59f5d3169e94ff8c01a7eb9R56
A EoClientCertAuthChallengeHandler is created in the constructor of AzureActiveDirectoryWebViewClient.
Do you think it would be better to have a corresponding method in AzureActiveDirectoryWebViewClient for starting the discovery?

Reply:
I actually think we may want to start / stop the discovery when there is a client cert request. I left a comment in the ctor there.